### PR TITLE
feat(api): create layout endpoint

### PR DESCRIPTION
**Pull request by p-fernandez**
### What change does this PR introduce?

Adds creation layout endpoint.

### Why was this change needed?

Unblocks development of the UI.

### CODE REVIEW

File: apps/api/src/app/layouts/dtos/layout.dto.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  contentType: string;
+
+  @ApiPropertyOptional()
+  variables?: ITemplateVariable[];
+
+  @ApiProperty()
+  isDefault: boolean;
+
+  @ApiProperty()
+  isDeleted: boolean;
```
Comment (ainouzgali) [inline apps/api/src/app/layouts/dtos/layout.dto.ts:45]: I think isDeleted could be removed. The plugin adds 'deleted' field

File: apps/api/src/app/layouts/layouts.controller.ts [hunk 2]
```diff
@@ -1,13 +1,49 @@
...
+    @Body() body: CreateLayoutRequestDto
+  ): Promise<CreateLayoutResponseDto> {
+    const layout = await this.createLayoutUseCase.execute(
+      CreateLayoutCommand.create({
+        environmentId: user.environmentId,
+        name: body.name,
+        organizationId: user.organizationId,
+        userId: user._id,
+        content: body.content,
+        variables: body.variables,
```
Comment (ainouzgali) [inline apps/api/src/app/layouts/layouts.controller.ts:40]: About the variables, I think they should be calculated inside the create usecase, like we do in [create template](https://github.com/novuhq/novu/blob/next/apps/api/src/app/notification-template/usecases/create-notification-template/create-notification-template.usecase.ts#L27)
But we can go over it or I can do it if you want

File: libs/shared/src/types/layout/index.ts [hunk 3]
```diff
@@ -0,0 +1,2 @@
+export type LayoutId = string;
+export type LayoutName = string;
```
Comment (ainouzgali) [inline libs/shared/src/types/layout/index.ts:2]: Should LayoutContent type also be added here as well?

**Review by ainouzgali — APPROVED** [review]

File: apps/api/src/app/layouts/dtos/layout.dto.ts (hunk 1 above)
Comment (p-fernandez) [reply apps/api/src/app/layouts/dtos/layout.dto.ts:45]: DTO is the base to return in the response so you would need it in other endpoints when returning the full entity (get layout, list layouts, etc).

File: apps/api/src/app/layouts/layouts.controller.ts (hunk 2 above)
Comment (p-fernandez) [reply apps/api/src/app/layouts/layouts.controller.ts:40]: I followed the create message use case logic. I see.
So what should be sent in the body of the creation of the layout? In my head I thought you send the content in one variable with the placeholders for the variables and in another property the variables and their values.

File: apps/api/src/app/layouts/dtos/layout.dto.ts (hunk 1 above)
Comment (p-fernandez) [reply apps/api/src/app/layouts/dtos/layout.dto.ts:45]: Now I understood what you meant. I got hit in a different PR by that. I am doing a mapping in the PR for the GET endpoint and will address it there. Thanks for flagging it!

File: apps/api/src/app/layouts/layouts.controller.ts (hunk 2 above)
Comment (p-fernandez) [reply apps/api/src/app/layouts/layouts.controller.ts:40]: As agreed privately, let's address the use case of parsing variables from a string when a user consumes the endpoint in a different PR: https://linear.app/novu/issue/NV-1470/use-case-parse-variables-from-user-content

**Pull request merged**